Common/Securities/Volatility/BaseVolatilityModel.cs

```diff
@@ -69,5 +71,68 @@ DateTime utcTime
         {
             return Enumerable.Empty<HistoryRequest>();
         }
+
+        /// <summary>
+        /// Gets history requests required for warming up the greeks with the provided resolution
+        /// </summary>
+        /// <param name="security">Security to get history for</param>
+        /// <param name="utcTime">UTC time of the request (end time)</param>
+        /// <param name="resolution">Resolution of the security</param>
+        /// <param name="barCount">Number of bars to lookback for the start date</param>
+        /// <returns>Enumerable of history requests</returns>
+        /// <exception cref="InvalidOperationException">The <see cref="SubscriptionDataConfigProvider"/> has not been set</exception>
+        public IEnumerable<HistoryRequest> GetHistoryRequirements(
+            Security security, 
+            DateTime utcTime,
+            Resolution? resolution,
+            int barCount)
+        {
+            if (SubscriptionDataConfigProvider == null)
+            {
+                throw new InvalidOperationException(
+                    "BaseVolatilityModel.GetHistoryRequirements(): " +
+                    "SubscriptionDataConfigProvider was not set."
+                );
+            }
+
+            var configurations = SubscriptionDataConfigProvider
+                .GetSubscriptionDataConfigs(security.Symbol)
```

Comment:
Let's add `OrderBy(config => config.TickType)` to be explicit and deterministic on the data type we choose for the history request, which will make it continue using `TradeBar`, if **available** in it